src/system.rs

```diff
@@ -1357,6 +1426,93 @@ mod schema_system_tests {
         assert!(schema.is_ok());
     }
 
+    #[test]
+    fn schema_system_map_authority_with_invalid_inline_import_type_test() {
```

Comment:
Can you explain to me how this is invalid? There are no inline imports here. All of the imports that do exist appear to be perfectly fine.

Reply:
My bad. Test name should be changed to `schema_system_map_authority_with_invalid_type_import_test`.
The reason for this is we are importing `my_text` which has a type reference `my_string`. Here only `my_text` is being imported. 
This is related to an error I explained above.

Reply:
Okay, so `sample.isl` should not be able to use `my_string` by name, but it should be able to use `my_text` without importing all of `my_text`'s dependencies. The "transitivity" that we're concern about actually has to do with whether a type can be referenced directly within the scope of a particular schema document. (Essentially, it's about each schema having a namespace for type names instead of having a single, global namespace.)

Reply:
Thanks for the explanation! Makes sense. I will change accordingly and add both valid and invalid tests to make sure the behavior is correct.